README.md

```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
+echo 'eval "$(rbenv init -)"' >> ~/.bashrc
+eval "$(rbenv init -)"
+```
+
+After this, you can install ruby with:
+
+```bash
+rbenv install 2.4.1
+```
+
+And after that, install bundler:
+
+```bash
+gem install bundler --no-ri --no-rdoc
+```
+
+### Cloning and configuring the application
+
+The first necessary step is to fork the following repositories:
+
+* [ManageIQ Core](https://github.com/ManageIQ/manageiq)
+* [ManageIQ UI Classic](https://github.com/ManageIQ/manageiq-ui-classic)
+* [ManageIQ Schema](https://github.com/ManageIQ/manageiq-schema)
+* [ManageIQ API](https://github.com/ManageIQ/manageiq-api)
+* [ManageIQ Hawkular Provider](https://github.com/ManageIQ/manageiq-providers-hawkular)
+* [Hawkular Ruby Client](https://github.com/hawkular/hawkular-client-ruby)
+
+Then, we clone them:
+
+```bash
+mkdir ~/ManageIQ && cd ~/ManageIQ
+git clone git@github.com:<your-username>/manageiq
+git clone git@github.com:<your-username>/manageiq-ui-classic
+git clone git@github.com:<your-username>/manageiq-schema
+git clone git@github.com:<your-username>/manageiq-api
+git clone git@github.com:<your-username>/manageiq-providers-hawkular
+git clone git@github.com:<your-username>/hawkular-client-ruby
+```
+
+And then we add the upstream remotes to fetch changes on the main codebase:
+
+```bash
+cd ~/ManageIQ/manageiq
+git remote add upstream git@github.com:ManageIQ/manageiq
+cd ..
+
+cd manageiq-ui-classic
+git remote add upstream git@github.com:ManageIQ/manageiq-ui-classic
+cd ..
+
+cd manageiq-schema
+git remote add upstream git@github.com:ManageIQ/manageiq-schema
+cd ..
+
+cd manageiq-api
+git remote add upstream git@github.com:ManageIQ/manageiq-api
+cd ..
+
+cd manageiq-providers-hawkular
+git remote add upstream git@github.com:ManageIQ/manageiq-providers-hawkular
+cd ..
+
+cd hawkular-client-ruby
+git remote add upstream git@github.com:hawkular/hawkular-client-ruby
+cd ..
+```
+
+Now we need to make ManageIQ use our vendored repositories instead of the local
+clone it does with Bundler. Create a file on `bundler.d` on the core repository
+called `vendor.rb` (any name is OK, but we're going for consistency here).
+
+This file is our addition to the Gemfile where we can override some gems and
+add some other development dependencies without messing with the main
+repository. Its contents are the following:
+
+```ruby
+override_gem name, path: File.expand_path("../../manageiq", __dir__)
+override_gem name, path: File.expand_path("../../manageiq-ui-classic", __dir__)
+override_gem name, path: File.expand_path("../../manageiq-schema", __dir__)
+override_gem name, path: File.expand_path("../../manageiq-api", __dir__)
+override_gem name, path: File.expand_path("../../manageiq-providers-hawkular", __dir__)
+```
+
+We also recommend some other dependencies that might come in hand for debugging:
+
+```ruby
+gem 'pry'
+gem 'pry-byebug'
+```
+
+The next step is to configure each repository:
+
+```bash
+cd ~/ManageIQ/manageiq
+./bin/setup
```

Comment:
isn't enough to run only this `bin/setup`? The other ones aren't needed unless you are running tests. But in order to run tests, some extra setup is required.

Reply:
What kind of extra setup?

Reply:
This should be: http://www.manageiq.org/docs/guides/developer_setup/plugins#running-tests

Reply:
Afaik the `spec:setup` task is also run when you run `bin/setup`, so we should be covered there.